tensorflow/compiler/tf2tensorrt/convert/ops/quantization_ops.h

```diff
@@ -0,0 +1,76 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+#ifndef TENSORFLOW_COMPILER_TF2TENSORRT_CONVERT_OPS_QUANTIZATION_OPS_H_
+#define TENSORFLOW_COMPILER_TF2TENSORRT_CONVERT_OPS_QUANTIZATION_OPS_H_
+#if GOOGLE_CUDA && GOOGLE_TENSORRT
+
+#include "absl/strings/str_format.h"
+#include "absl/strings/str_join.h"
+#include "tensorflow/compiler/tf2tensorrt/convert/convert_nodes.h"
+#include "tensorflow/core/lib/core/status.h"
+
+namespace tensorflow {
+namespace tensorrt {
+namespace convert {
+
+constexpr std::array<const char*, 4> kQuantizationOpNames = {
+    "QuantizeAndDequantizeV2",
+    "QuantizeAndDequantizeV3",
+    "FakeQuantWithMinMaxVars",
+    "FakeQuantWithMinMaxArgs",
+};
+
+// Operations with supported conversion to Q/DQ ops in TensorRT explicit
+// precision mode.
+constexpr std::array<const char*, 1> kExplicitQuantizationOpNames = {
+    "QuantizeAndDequantizeV2",
+};
+
+// Contains two scaling factors for quantization and dequantization
+// respectively. A shift factor is omitted as TensorRT only supports symmetric
+// quantization.
+template <typename T, size_t N>
+struct QuantizationScales {
+  std::array<T, N> quantize_scale;
+  std::array<T, N> dequantize_scale;
+};
+
+// In TensorRT 7 and 8, only uniform tensor scaling is supported for
```

Comment:
Shall we s/8/8.0/, because for > 8.0, per channel quantization is supported?

Reply:
For activations, only uniform scaling is supported.
For weights, per channel is supported for TRT >=8.0.